.github/workflows/R-CMD-check-dev.yaml

```diff
@@ -0,0 +1,96 @@
+on:
```

Comment:
Why do we need a new workflow file? What does it contain? How do we maintain it as new R-CMD-check.yaml versions come in? Is it based on R-CMD-check.yaml? In that case do we need a commit with a clear diff?

Reply:
- Splitting workflows into multiple files (one per purpose) seems better than one big workflow with lots of branches.
- Dependencies are installed from GitHub via `remotes::install_dev()`, caching is enabled. Checks on push/PR and also once per day, it seems difficult to trigger a check on upstream changes
- This can be part of https://github.com/r-lib/actions, needs some exposure before that
- Yes
- We can have that commit in r-lib/actions if necessary